### ISSUE TITLE
Feature Request: Develop an Automatic Report Generator with Scheduler 

### ISSUE THREAD

**Issue by ScoutLeaderDid** [pete-mc/Summit#78]
Re: enhancement

It would be beneficial if you can develop an automatic report generator with email scheduler. 

Options in this feature/module would;
1. Select Report
2. Select who to email to - predefined or adhoc
3. Select frequency, reoccurring days, say send OAS Summary report, every fortnight on a Thursday morning at 9 am

All the reports by you Pete and Topo are brilliant, By having a automatic scheduler, it saves the leaders having to log in and run one of at least 2-4 reports every couple of times per month. Ultimately the idea is the reports lands in the leader email box. He/She will simply open and print the PDF with minimal effort. The report can either be posted on a notice board or used when doing badge work with the Scout/s.
 
Regards,
Did

**Comment by pete-mc** [comment]
Yeah, I agree. I also wanted to have alerting for things like pending approvals and maybe a combination of toast notifications and emails depending on the user settings. This will require some sort of "Back End" which I am working through some ideas that would still keep the data within the Scouts systems to reduce and security issues or concerns. 

I am trying to work out if every branch uses Office 365 because if they do I have a few options of using its inbuilt features to manage these sorts of back-end tasks and simultaneously keeping it within Scouts controlled areas.

**Comment by jamesanderson16** [comment]
@pete-mc & @ScoutLeaderDid  I think most scout branches are now using a singular Microsoft 365 system across australia due to Microsoft killing what the licences that the branches use to use (they are now using an Education Licence)
